torch/_inductor/codegen/cpp_bmm_template.py

```diff
@@ -0,0 +1,171 @@
+# mypy: allow-untyped-defs
+from typing import Callable, List, Optional
+
+from .. import ir
+
+from ..select_algorithm import DataProcessorTemplateWrapper
+from .cpp_gemm_template import CppPackedGemmTemplate, GEMM_TEMPLATE, MICROKERNEL_DEF
+
+from .cpp_template_kernel import CppTemplateKernel
+from .cpp_utils import GemmBlocking
+
+GEMM_SINGLE_THREAD_MM_STUB = r"""
+void single_thread_mm(
+    const {{micro_gemm.get_common_options()['input_t']}}* X,
+    const {{micro_gemm.get_common_options()['input_t']}}* W,
+    {{micro_gemm.get_common_options()['input_t']}}* Y
```

Comment:
Using `output_t` represents the output type of the microkernel, not the output of the GEMM. For now I'll keep using `input_t`, unless you have a different suggestion?

Reply:
Then it feels wrong since `single_thread_mm` and `threaded_mm` should have same input/output data type as gemm template instead of micro gemm. Can get the data type of "BX", "BW" and “BY”, save it in the "options" and using it here.

Reply:
Changed as you suggested.